src/docs/markdown/automatic-https.md

```diff
@@ -220,3 +222,83 @@ Before attempting any ACME transactions, Caddy will test the configured storage
 Caddy can obtain and manage wildcard certificates when it is configured to serve a site with a qualifying wildcard name. A site name qualifies for a wildcard if only its left-most domain label is a wildcard. For example, `*.example.com` qualifies, but these do not: `sub.*.example.com`, `foo*.example.com`, `*bar.example.com`, and `*.*.example.com`.
 
 To get a wildcard from Let's Encrypt, you simply need to enable the [DNS challenge](#dns-challenge) and use a wildcard domain in your config. We recommend using wildcards only when you have so many subdomains that you would encounter CA rate limits trying to obtain certificates for them all.
+
+
+
+## Examples
+
+Caddy implicitly uses the HTTPS port (default 443) for your [server addresses](/docs/conventions#network-addresses) that don't assign an explicit port (which would disable automatic HTTPS). 
+
+The global setting `auto_https` has two values:
+- `disable_redirects` adds an implicit HTTP port redirect.
+- `off` disables automatic HTTPS, default implicit port changes to HTTP for all server addresses.
+
+| auto_https        | HTTP        | HTTPS       |
+|-------------------|-------------|-------------|
+| *default*         | redirects   | connects    |
+| disable_redirects | unreachable | connects    |
+| off               | connects    | unreachable |
+
+`unreachable` is the default without additional configuration when using the related `auto_https` setting.
+`connects` is the implicitly assigned port.
+
+
+### Disabling automatic HTTPS
+
+For local development environments, you can prevent serving via HTTPS by providing an explicit port assignment(disable per server address) or using the global `auto_https off` setting which will change Caddy's implicit port to be the HTTP port (default 80) globally.
+
+```caddy
+{
+  auto_https off
+}
+
+# Have caddy implicitly use the HTTP port
+localhost {
+  root * /usr/share/caddy
+
+  file_server
+}
+```
+
+```caddy
+# Alternatively, provide an explicit port
+localhost:9000 {
+  root * /usr/share/caddy
+
+  file_server
+}
+```
+
+
+### HTTP and HTTPS without redirect
+
+If you want to serve content through both HTTP and HTTPS ports without HTTP redirects, your server address cannot rely on a single implicit port and you must explicitly declare the intent to listen from both ports.
+
+To do so, you can [map several addresses to a site block as a list](/docs/caddyfile/concepts#addresses) separated with `,`:
```

Comment:
```suggestion
If you wish to serve content through both HTTP and HTTPS, without HTTP->HTTPS redirects, your site label cannot must explicitly declare the intent to listen on both ports. To do so, [specify multiple site labels](/docs/caddyfile/concepts#addresses), separated by a comma or whitespace.
```

Reply:
> site label 

I did not see this terminology used in the docs yet? I've seen "Network Addresses", "Addresses", and their corresponding Block "site block". I do prefer site label though, in a few places I've prepended "server" when mentioning addresses for added clarity. Since it's not always for websites(could be for an API), is "service address" better?

[Conventions - Network Addresses](https://caddyserver.com/docs/conventions#network-addresses) tip section defines "Network Addresses" pretty well, additionally the [concepts section](https://caddyserver.com/docs/caddyfile/concepts#addresses) visual breakdown has a legend calling it a "site address", and in the "Addresses" section wildcard matching refers to a label as part of a hostname.

[URI](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier) also seems to be an apt description I think?

> cannot must

I think the "cannot" was meant to be removed?

Reply:
I have pushed changes removing the "cannot", and renamed "site label" and "server addresses" to "site address/es".

Reply:
There's an unpublished `spec.md` document that you can find in here that talks a lot about labels. That was the terminology used in Caddy v1 and I may have missed the decision where @mholt switched the terminology... but I'm not sure.

As you said, the wildcard bit specifically talks about one segment of the address being a label, which follows from the `{labels.*}` placeholders that exist. Maybe label isn't the best word here.

👍 for the "cannot" typo